### ISSUE TITLE
Nested folders showing up as 1 item after using and removing New File extension (SOLVED)

### ISSUE THREAD

**Issue by tibberous** [microsoft/vscode#210849]
Type: <b>Bug</b>

I installed an Advanced New File extension which I have since removed. The extension allows you to make a new file and specify a path, and if that path doesn't exist it recursively creates it.

So you can make a new file named (workspace folder)\a\b\c.txt , and if a\b\ doesn't exist, it will create the folder a\, then inside it create the folder b\, then make the file c.txt. I wish this was the default behavior of Visual Studio Code.

However, now my workspace contains one directory called a \ b \ c and it all apears on 1 line. I can't open b to see the folder c, it lists it in the workspace as one item.

And even after removing the extension, I still have the bug. You can view a screenshot here: https://trentontompkins.com/screenshot.jpg (it would be nice if your bug report gave the option to include the current screen)

VS Code version: Code 1.88.1 (e170252f762678dec6ca2cc69aba1570769a5d39, 2024-04-10T17:41:02.734Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.79GB (7.97GB free)|
|Process Argv|--crash-reporter-id 90255ea8-dddc-478b-a9a4-15ab5525a676|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-html-css|ecm|2.0.9
prettier-vscode|esb|10.4.0
code-runner|for|0.12.2
git-graph|mhu|1.30.0
vscode-docker|ms-|1.29.0
remote-wsl|ms-|0.88.0
rainbow-bracket|tal|1.0.2
highlight-logical-line|use|0.1.0
vscplus|Vre|1.1.5
vscode-icons|vsc|12.7.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
vscaac:30438847
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
d34g3935:30971562
fegfb526:30981948
bg6jg535:30979843
ccp2r3:30993541
dsvsc020:30976470
pythonait:31006305
gee8j676:31009558
dsvsc021:30996838
da93g388:31013173
pythoncenvpt:31022790

```

</details>

**Comment by gjsjohnmurray** [comment]
Sounds like the `explorer.compactFolders` setting, described at https://code.visualstudio.com/updates/v1_41#_compact-folders-in-explorer

Re screenshots, after Report Issue you should get a browser page where you can either edit the issue description and paste (or drag/drop) a screenshot, or add it on a comment on the issue.

**Comment by tibberous** [comment]
Omg! You are 100% correct!

As soon as I changed the setting the "bug" was fixed - I've heard the joke, "it's not a bug, it's a feature, but I've never seen that literally be true.

Spent an hour on this, it was viscerally bothering me when I woke up this morning. Because it wasn't something I *needed* to do, but I wanted to use those plugins, but now if they introduced some weird bug in my project. 

It might be worth thinking about whether the default value for that setting should be true. Because the default behavior is what you'd expect, and now that I know there is a setting, I can see why that option could be nice in some instances. But I can't think of many. So if it was off by default - I think that would be maybe better on the whole? But thank you so much - you helped me a ton, and I hope this post can help other people understand the default behavior.
